lib/shared/store.js

```diff
@@ -319,24 +328,19 @@ const storeReducer = (state = DEFAULT_STATE, action) => {
     // place where all the image extension / format handling
     // takes place, to avoid having to check 2+ locations with different logic
     case ACTIONS.SELECT_IMAGE: {
-      if (!action.data.path) {
-        throw errors.createError({
-          title: 'Missing image path'
-        })
-      }
+      const noNilFields = [
```

Comment:
Same here – let's try to avoid allocating things that are constant over and over